Swiften/MIX/MIXRegistry.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#include <Swiften/MIX/MIXRegistry.h>
+
+namespace Swift {
+
+MIXRegistry::MIXRegistry(const JID& ownJID, IQRouter* iqRouter, XMPPRoster* xmppRoster) : ownJID_(ownJID), iqRouter_(iqRouter), xmppRoster_(xmppRoster), rosterPopulated_(false) {
+    xmppRoster_->onInitialRosterPopulated.connect(boost::bind(&MIXRegistry::handleRosterPopulated, this));
+    xmppRoster_->onJIDAdded.connect(boost::bind(&MIXRegistry::handleJIDAdded, this, _1));
+    xmppRoster_->onJIDRemoved.connect(boost::bind(&MIXRegistry::handleJIDRemoved, this, _1));
+}
+
+MIXRegistry::~MIXRegistry() {
+
+}
+
+void MIXRegistry::joinChannel(const JID& channelJID, const std::unordered_set<std::string>& nodes) {
```

Comment:
This should probably check if a channel with `channelJID` is already on the roster and log a warning instead of issuing a join.